File: pintia/pat_basic/_1111.cpp
#include <iostream>
#include <unordered_map>
#include <cstring>
#include <algorithm>

using namespace std;

unordered_map<string, int> M;
int n;
char dt[20], dt_rev[20];

void init() {
    M["Jan"] = 1;
    M["Feb"] = 2;
    M["Mar"] = 3;
    M["Apr"] = 4;
    M["May"] = 5;
    M["Jun"] = 6;
    M["Jul"] = 7;
    M["Aug"] = 8;
    M["Sep"] = 9;
    M["Oct"] = 10;
    M["Nov"] = 11;
    M["Dec"] = 12;
}

int main() {
    cin >> n;
    init();
    string sm, ign;
    int d, y;
    while (n--) {
        cin >> sm >> d >> ign >> y;
        int m = M[sm];
        sprintf(dt, "%04d%02d%02d", y, m, d);
        strcpy(dt_rev, dt);
        int len = strlen(dt);
        reverse(dt_rev, dt_rev + len);
        printf(strcmp(dt, dt_rev) ? "N" : "Y");
        printf(" %s\n", dt);
    }

    return 0;
}

File: pintia/pat_basic/_1111.java
import java.util.*;
import java.io.*;

public class _1111 {
    static Map<String, Integer> mp = new HashMap<>();
    static {
        mp.put("Jan", 1);
        mp.put("Feb", 2);
        mp.put("Mar", 3);
        mp.put("Apr", 4);
        mp.put("May", 5);
        mp.put("Jun", 6);
        mp.put("Jul", 7);
        mp.put("Aug", 8);
        mp.put("Sep", 9);
        mp.put("Oct", 10);
        mp.put("Nov", 11);
        mp.put("Dec", 12);
    }

    public static void main(String[] args) throws IOException {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        in.nextLine();
        while (n-- > 0) {
            int mm = mp.get(in.next());
            String s = in.next();
            int dd = Integer.parseInt(s.substring(0, s.length() - 1));
            int yy = in.nextInt();
            String dt = String.format("%04d%02d%02d", yy, mm, dd);
            System.out.printf("%c %s\n", dt.equals(new StringBuilder(dt).reverse().toString()) ? 'Y' : 'N', dt);
        }
    }
}
